Bottom: 2b620c03c80f414b832aec6d7b3ee8b3f9f77c0d
Top:    b2b6f37ddfa12e1ffebfe62957952aaa76f3afda
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 01:33:10 +0000

polygon1.c: Fix Collect routine

Not 100% sure about this, but it seems to fix a nasty
case regarding incorrect geometry created in the following
test geometry:

Element["" "" "" "unknown" 1325.00mil 1070.00mil -175.00mil -105.00mil 0 100 ""]
(
  Pad[-85.00mil -39.00mil -79.00mil -39.00mil 34.00mil 30.00mil 40.00mil "2" "2" "square"]
)

Layer(1 "component")
(
  Line[1245.00mil 1030.00mil 1195.00mil 1030.00mil 12.00mil 20.00mil "clearline"]
  Polygon("clearpoly")
  (
    [5.00mil 5.00mil] [4195.00mil 5.00mil] [4195.00mil 2695.00mil] [5.00mil 2695.00mil]
  )
)


---

diff --git a/src/polygon1.c b/src/polygon1.c
index ef82f37..567a473 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1695,7 +1695,7 @@ Collect (jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes,
       if (s_rule (cur, &dir) && cur->Flags.mark == 0)
 	Collect1 (e, cur, dir, contours, holes, j_rule);
       other = cur;
-      if ((other->cvc_prev && jump (&other, &dir, j_rule)))
+      if (other->cvc_prev)
 	Collect1 (e, other, dir, contours, holes, j_rule);
     }
   while ((cur = cur->next) != &a->head);
